tensorflow_probability/python/distributions/gev_test.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2020 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Tests for GEV."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+import numpy as np
+from scipy import stats
+
+import tensorflow.compat.v1 as tf1
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+
+from tensorflow_probability.python.internal import test_util
+
+tfd = tfp.distributions
+
+
+class _GEVTest(object):
```

Comment:
Since I had all those concerns about NaN gradients, would you mind adding a test for those? See https://github.com/tensorflow/probability/blob/612a9cc27cc96f6dad433d604c8255eb9daf5e2e/tensorflow_probability/python/distributions/generalized_normal_test.py#L168-L186 for an example. I especially care about the branch points of concentration=1, 0.5, 0 but other spots may be good to check too.